app/models/embeddable/embeddable_plugin.rb

```diff
@@ -10,7 +10,7 @@ def self.table_name
     attr_accessible :plugin, :approved_script_id, :description, :author_data,
     :is_full_width, :is_hidden, :component_label
 
-    belongs_to :plugin, autosave: true
+    has_one :plugin, as: :plugin_scope
```

Comment:
I will leave it to @knowuh to review this part - I'm still not confident in my knowledge of rails model relationship magic.